### PR TITLE
Add ability to rename columns

### PR DESCRIPTION
**Pull request by aforward**
I am looking to add support for renaming columns in ecto migrations.

Here is the documentation I referred to regarding the Postgres and MySQL syntax.

http://www.postgresql.org/docs/9.4/static/sql-altertable.html

https://dev.mysql.com/doc/refman/5.1/en/alter-table.html

### CODE REVIEW

**Comment by aforward** [comment]
I will have to look more closely at the integration tests to ensure the RENAME syntax is correct

```
** (Postgrex.Error) ERROR (syntax_error): syntax error at or near "RENAME"
    (ecto) lib/ecto/adapters/sql.ex:210: Ecto.Adapters.SQL.log_and_check/7
    (ecto) lib/ecto/adapters/postgres.ex:59: Ecto.Adapters.Postgres.execute_ddl/3
    integration_test/support/migration.exs:77: Ecto.Integration.Migration.create_users_table/0
    integration_test/support/migration.exs:23: Ecto.Integration.Migration.change/0
    (stdlib) timer.erl:194: :timer.tc/3
    (ecto) lib/ecto/migration/runner.ex:22: Ecto.Migration.Runner.run/6
    (ecto) lib/ecto/migrator.ex:113: Ecto.Migrator.attempt/6
    (ecto) lib/ecto/migrator.ex:63: anonymous fn/4 in Ecto.Migrator.do_up/4
```

File: integration_test/support/migration.exs [hunk 1]
```diff
@@ -80,6 +81,10 @@ defmodule Ecto.Integration.Migration do
       timestamps
     end
 
+    alter table(:users) do
```
Comment (aforward) [inline integration_test/support/migration.exs]: Renaming a column is not an "action" as defined in
http://www.postgresql.org/docs/9.4/static/sql-altertable.html

So it currently must be run in a separate alter table(:xxx) call.  Is something like that acceptable?  Or, should we detect it in the backend and ensure two multiple alter tables are executed when a rename is used.

Comment (josevalim) [inline integration_test/support/migration.exs]: We don't necessarily need to do separate calls but we do need to generate a special SQL for postgres. So we need to traverse the alter table instructions, finding all rename commands, and then group all queries by using ";" as separator. If that doesn't work, we can try BEGIN END too:

```sql
BEGIN
  ALTER TABLE foo rename 1;
  ALTER TABLE foo rename 2;
  ALTER TABLE foo others
END;
```

Hopefully one of those ideas work. :)

**Comment by josevalim** [comment]
Ping. :)

File: integration_test/support/migration.exs (hunk 1 above)
Comment (aforward) [inline integration_test/support/migration.exs]: Will do.  I'll attack this on the weekend.

**Comment by josevalim** [comment]
Closing this in favor of #807. Thanks!

**Comment by aforward** [comment]
Sorry I couldn't pick up the slack here.

**Pull request closed without merging**